packit_service/worker/handlers/distgit.py

```diff
@@ -113,6 +115,8 @@ class AbortProposeDownstream(Exception):
 @run_for_comment(command="propose-update")  # deprecated
 @run_for_check_rerun(prefix="propose-downstream")
 @reacts_to(event=ReleaseEvent)
+@reacts_to(event=ReleaseGitlabEvent)
+@reacts_to(event=TagPushGitlabEvent)
```

Comment:
I am not familiar with GitLab enough to know what would be the best experience here: if Packit should react to both or to just one of these (or even make it configurable?)

Does anyone know if tags & releases work the same way on GitLab as they work on GitHub?

Reply:
I would keep it consistent with GitHub and that would be just the `ReleaseGitlabEvent`.

Reply:
Thinking about it, this could also cause some issues with `copy_upstream_release_description` or whatever that setting for populating changelog is called.

Reply:
that sounds perfect! let users configure what they send to us :)

Reply:
> that sounds perfect! let users configure what they send to us :)

Given that we have `copy_upstream_release_description`, this could be an issue with the tags, so unless they use it, I guess those two triggers could be interchangeable, but it sounds a bit fragile tbf

Reply:
If I should decide I will go with the 2nd, but I do not really understand the drawbacks of this. So I wait for someone to tell me what to do 😅. Thanks!

Reply:
Now I have seen the Tomas preference *let users configure what they send to us*. 
Then, I try to update the documentation.

Reply:
tl;dr if you kept both `@reacts_to` and ticked both tag and releases in gitlab's webhook settings, we would get both of the events, and if tag push and release was done in a close proximity of each other, we could end up with a race condition (and very likely duplicit PRs on downstream)

Reply:
Personally I think that if we end up duplicating the PR in distgit, upon a wrong user configuration, this is not too bad. But the race condition could lead to bad exceptions which could be more difficult to understand.

So, should we trust our users? Should we keep both events? Or should we just keep the `Release` event and maybe create another card to study a better solution to introduce the `TagPush` event?

Reply:
I think it would make sense to react just to releases right now and open some low-prio card for the tag push (on both github and gitlab), but as of now I am not aware of anyone requesting that, simpler compared to release, solution.

I'm also against going with tag push to prod rightaway. With `release` I am not worried that much.

Reply:
https://github.com/packit/packit-service/issues/1785